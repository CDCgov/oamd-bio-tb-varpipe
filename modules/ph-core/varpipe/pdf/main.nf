process VARPIPE_PDF {

    tag "${meta.id}"
    label 'process_single'

    container 'oamd-bio-crc:0.12.5_822d693_v3'

    input:
    tuple val(meta), path(summary)

    output:
    tuple val(meta), path("*_report.pdf"), emit: report
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'pdf_print.py'

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_report.pdf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | cut -d' ' -f2)
    END_VERSIONS
    """
}
