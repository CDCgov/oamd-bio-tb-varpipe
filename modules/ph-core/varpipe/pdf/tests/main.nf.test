nextflow_process {

    name "Test Process VARPIPE_PDF"
    script "../main.nf"
    process "VARPIPE_PDF"

    tag "modules"
    tag "modules_phcore"
    tag "varpipe"
    tag "varpipe/pdf"

    test("varpipe pdf report [summary]") {

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test1_summary.txt')
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.report.size() > 0 },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }

    test("varpipe pdf report [summary] -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test1_summary.txt')
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
