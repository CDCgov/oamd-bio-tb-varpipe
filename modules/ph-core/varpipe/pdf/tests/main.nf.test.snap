{
    "varpipe pdf report [summary] -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_report.pdf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ab080f45d92d50e9010184534d46f0c7"
                ],
                "report": [
                    [
                        {
                            "id": "test"
                        },
                        "test_report.pdf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ab080f45d92d50e9010184534d46f0c7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-07T13:37:54.999508164"
    },
    "varpipe pdf report [summary]": {
        "content": [
            [
                "versions.yml:md5,b2225a03d1f31d7a0958bd9de8431601"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-07T13:37:47.959880705"
    }
}
